diff -rupN partclone-0.2.88/partclone-0.2.88.patch partclone-0.2.88-new/partclone-0.2.88.patch
--- partclone-0.2.88/partclone-0.2.88.patch	1969-12-31 19:00:00.000000000 -0500
+++ partclone-0.2.88-new/partclone-0.2.88.patch	2016-05-29 16:02:10.439369965 -0400
@@ -0,0 +1,65 @@
+diff -rupN partclone-0.2.88-original/src/progress.c partclone-0.2.88-new/src/progress.c
+--- partclone-0.2.88-original/src/progress.c	2016-02-18 22:15:26.000000000 -0500
++++ partclone-0.2.88-new/src/progress.c	2016-04-01 10:59:04.668998720 -0400
+@@ -182,6 +182,45 @@ static void calculate_speed(struct progr
+     strncpy(prog_stat->Rformated, Rformated, sizeof(prog_stat->Rformated));
+ }
+ 
++/// convert the size
++char* filesize_conv(double size, char *buf, size_t len) {
++    int i = 0;
++    const char* units[] = {"iB","KiB","MiB","GiB","TiB","PiB","EiB","ZiB","YiB"};
++    while (size > 1024) {
++        size /= 1024;
++        i++;
++    }
++    snprintf(buf, len, "%.*f %s", i, size, units[i]);
++    return buf;
++}
++
++/// write data to log file for status/progress info
++void fogLogStatusFile(struct progress_bar *prog,unsigned long long current) {
++    prog_stat_t prog_stat;
++    FILE *fog_log;
++    fog_log = fopen("/var/log/status.fog","w");
++    if (fog_log == NULL) {
++        fprintf(stderr,"Error opening file %s","/var/log/status.log");
++        exit(0);
++    }
++    double totalsize=prog->block_size * prog->total;
++    double currentsize=prog->block_size * current;
++    int max_len=15;
++    char total_str[max_len];
++    char current_str[max_len];
++    char buf[max_len];
++    sprintf(total_str,filesize_conv(totalsize,buf,max_len));
++    sprintf(current_str,filesize_conv(currentsize,buf,max_len));
++    fprintf(fog_log,"%6.2f%s@",prog_stat.speed,prog_stat.speed_unit);
++    fprintf(fog_log,"%s@",prog_stat.Eformated);
++    fprintf(fog_log,"%s@",prog_stat.Rformated);
++    fprintf(fog_log,"%s@",current_str);
++    fprintf(fog_log,"%s@",total_str);
++    fprintf(fog_log,"%6.2f@",prog_stat.percent);
++    fprintf(fog_log,"%f\n",totalsize);
++    fclose(fog_log);
++}
++
+ /// update information at progress bar
+ extern void progress_update(struct progress_bar *prog, unsigned long long copied, unsigned long long current, int done)
+ {
+@@ -203,6 +241,7 @@ extern void progress_update(struct progr
+ 	    fprintf(stderr, _(", %6.2f%s/min,"), prog_stat.speed, prog_stat.speed_unit);
+ 	if(prog->flag == IO)
+ 	    fprintf(stderr, "\n\r%80c\rcurrent block: %10lld, total block: %10lld, Complete: %6.2f%%%s\r", clear_buf, current, prog->total, prog_stat.total_percent, "\x1b[A");
++    fogLogStatusFile(prog, current);
+     } else {
+ 	setlocale(LC_ALL, "");
+ 	bindtextdomain(PACKAGE, LOCALEDIR);
+@@ -278,6 +317,7 @@ extern void Ncurses_progress_update(stru
+ 	wrefresh(p_win);
+         wrefresh(bar_win);
+         wrefresh(tbar_win);
++    fogLogStatusFile(prog, current);
+     } else {
+         mvwprintw(p_win, 0, 0, _("Total Time: %s Remaining: %s "), prog_stat.Eformated, prog_stat.Rformated);
+ 	if((prog->flag == IO) || (prog->flag == NO_BLOCK_DETAIL))
diff -rupN partclone-0.2.88/src/Makefile.am partclone-0.2.88-new/src/Makefile.am
--- partclone-0.2.88/src/Makefile.am	2016-02-18 22:15:26.000000000 -0500
+++ partclone-0.2.88-new/src/Makefile.am	2016-05-29 16:40:06.679348257 -0400
@@ -22,7 +22,7 @@ EXFATFS_SOURCE=exfat/cluster.c exfat/utf
 F2FS_SOURCE=f2fs/fsck.c f2fs/libf2fs.c f2fs/fsck.h f2fs/mount.c f2fs/f2fs_fs.h f2fs/list.h f2fs/f2fs.h
 
 if ENABLE_STATIC
-AM_LDFLAGS=-static
+AM_LDFLAGS=
 endif
 
 if ENABLE_TINFO
diff -rupN partclone-0.2.88/src/Makefile.in partclone-0.2.88-new/src/Makefile.in
--- partclone-0.2.88/src/Makefile.in	2016-02-18 22:15:26.000000000 -0500
+++ partclone-0.2.88-new/src/Makefile.in	2016-05-29 16:15:37.039362272 -0400
@@ -749,6 +749,7 @@ xfs/atomic.h xfs/libxfs_priv.h xfs/xfs_a
 EXFATFS_SOURCE = exfat/cluster.c exfat/utf.c exfat/utils.c exfat/lookup.c exfat/io.c exfat/log.c exfat/node.c exfat/mount.c exfat/time.c
 F2FS_SOURCE = f2fs/fsck.c f2fs/libf2fs.c f2fs/fsck.h f2fs/mount.c f2fs/f2fs_fs.h f2fs/list.h f2fs/f2fs.h
 @ENABLE_STATIC_TRUE@AM_LDFLAGS = -static
+@ENABLE_STATIC_FALSE@AM_LDFLAGS =
 AM_CFLAGS = -D_FILE_OFFSET_BITS=64 $(am__append_2) $(am__append_3)
 partclone_info_SOURCES = info.c partclone.c partclone.h fs_common.h
 partclone_restore_SOURCES = main.c partclone.c progress.c ddclone.c ddclone.h partclone.h progress.h gettext.h
diff -rupN partclone-0.2.88/src/progress.c partclone-0.2.88-new/src/progress.c
--- partclone-0.2.88/src/progress.c	2016-02-18 22:15:26.000000000 -0500
+++ partclone-0.2.88-new/src/progress.c	2016-05-29 16:02:10.443369965 -0400
@@ -182,6 +182,47 @@ static void calculate_speed(struct progr
     strncpy(prog_stat->Rformated, Rformated, sizeof(prog_stat->Rformated));
 }
 
+/// convert the size
+char* filesize_conv(double size, char *buf, size_t len) {
+    int i = 0;
+    const char* units[] = {"iB","KiB","MiB","GiB","TiB","PiB","EiB","ZiB","YiB"};
+    while (size > 1024) {
+        size /= 1024;
+        i++;
+    }
+    snprintf(buf, len, "%.*f %s", i, size, units[i]);
+    return buf;
+}
+
+/// write data to log file for status/progress info
+void fogLogStatusFile(struct progress_bar *prog,struct prog_stat_t *prog_stat,unsigned long long current) {
+    FILE *fog_log;
+    fog_log = fopen("/var/log/status.fog","w");
+    if (fog_log == NULL) {
+        fprintf(stderr,"Error opening file %s","/var/log/status.log");
+        exit(0);
+    }
+    double totalsize=prog->block_size * prog->total;
+    double currentsize=prog->block_size * current;
+    int max_len=15;
+    char total_str[max_len];
+    char current_str[max_len];
+    char buf[max_len];
+    sprintf(total_str,filesize_conv(totalsize,buf,max_len));
+    sprintf(current_str,filesize_conv(currentsize,buf,max_len));
+    fprintf(fog_log,"%6.2f%s@%s@%s@%s@%s@%6.2f@%f\n",
+            prog_stat->speed,
+            prog_stat->speed_unit,
+            prog_stat->Eformated,
+            prog_stat->Rformated,
+            current_str,
+            total_str,
+            prog_stat->percent,
+            totalsize
+    );
+    fclose(fog_log);
+}
+
 /// update information at progress bar
 extern void progress_update(struct progress_bar *prog, unsigned long long copied, unsigned long long current, int done)
 {
@@ -203,6 +244,7 @@ extern void progress_update(struct progr
 	    fprintf(stderr, _(", %6.2f%s/min,"), prog_stat.speed, prog_stat.speed_unit);
 	if(prog->flag == IO)
 	    fprintf(stderr, "\n\r%80c\rcurrent block: %10lld, total block: %10lld, Complete: %6.2f%%%s\r", clear_buf, current, prog->total, prog_stat.total_percent, "\x1b[A");
+    fogLogStatusFile(prog, &prog_stat, current);
     } else {
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
@@ -278,6 +320,7 @@ extern void Ncurses_progress_update(stru
 	wrefresh(p_win);
         wrefresh(bar_win);
         wrefresh(tbar_win);
+    fogLogStatusFile(prog, &prog_stat, current);
     } else {
         mvwprintw(p_win, 0, 0, _("Total Time: %s Remaining: %s "), prog_stat.Eformated, prog_stat.Rformated);
 	if((prog->flag == IO) || (prog->flag == NO_BLOCK_DETAIL))
