--- partclone-0.3.13_orig/src/main.c	2020-09-27 13:36:21.000000000 +0100
+++ partclone-0.3.13_fix/src/main.c	2020-09-27 23:45:10.000000000 +0100
@@ -718,8 +718,10 @@
 
 				if (opt.ignore_crc) {
 					read_offset += block_size;
-					if (++blocks_in_cs == blocks_per_cs)
+					if (++blocks_in_cs == blocks_per_cs) {
 						read_offset += cs_size;
+						blocks_in_cs = 0;
+					}
 					continue;
 				}
 
@@ -744,7 +746,7 @@
 
 				read_offset += block_size;
 			}
-			if (blocks_in_cs && blocks_per_cs && blocks_read < buffer_capacity &&
+			if (!opt.ignore_crc && blocks_in_cs && blocks_per_cs && blocks_read < buffer_capacity &&
 					(blocks_read % blocks_per_cs)) {
 
 			    log_mesg(1, 0, 0, debug, "check latest chunk's checksum covering %u blocks\n", blocks_in_cs);
