diff -Nur linux-4.19.90/drivers/scsi/storvsc_drv.c kernelsourcex86/drivers/scsi/storvsc_drv.c
--- linux-4.19.90/drivers/scsi/storvsc_drv.c    2019-12-17 20:22:00.000000000 -0500
+++ kernelsourcex86/drivers/scsi/storvsc_drv.c  2019-12-20 17:44:00.000000000 -0500
@@ -1435,7 +1435,12 @@
 	blk_queue_rq_timeout(sdevice->request_queue, (storvsc_timeout * HZ));
 
 	/* Ensure there are no gaps in presented sgls */
-	blk_queue_virt_boundary(sdevice->request_queue, PAGE_SIZE - 1);
+	if (PAGE_SIZE - 1 < 4096) {
+		blk_queue_virt_boundary(sdevice->request_queue, 4096);
+	}
+	else {
+		blk_queue_virt_boundary(sdevice->request_queue, PAGE_SIZE - 1);
+	}
 
 	sdevice->no_write_same = 1;
 
