diff -rupN linux-4.17_orig/drivers/scsi/storvsc_drv.c linux-4.17_new/drivers/scsi/storvsc_drv.c
--- linux-4.17_orig/drivers/scsi/storvsc_drv.c	2018-06-03 17:15:21.000000000 -0400
+++ linux-4.17_new/drivers/scsi/storvsc_drv.c	2018-06-05 17:00:54.908177156 -0400
@@ -1388,7 +1388,11 @@ static int storvsc_device_configure(stru
 	blk_queue_rq_timeout(sdevice->request_queue, (storvsc_timeout * HZ));
 
 	/* Ensure there are no gaps in presented sgls */
-	blk_queue_virt_boundary(sdevice->request_queue, PAGE_SIZE - 1);
+    if (PAGE_SIZE -1 < 4096) {
+        blk_queue_virt_boundary(sdevice->request_queue, 4096);
+    } else {
+        blk_queue_virt_boundary(sdevice->request_queue, PAGE_SIZE - 1);
+    }
 
 	sdevice->no_write_same = 1;
 
